src/wire/ipv4.rs

```diff
@@ -273,7 +273,7 @@ impl<T: AsRef<[u8]>> Packet<T> {
             Err(Error::Truncated)
         } else if self.header_len() as u16 > self.total_len() {
             Err(Error::Malformed)
-        } else if len < self.total_len() as usize {
+        } else if len < self.total_len() as usize && self.dont_frag() {
```

Comment:
This needs an added test.

Reply:
Can you elaborate? If the packet is not fragmented, the check goes on as before. If it is fragmented, I skip this check (and verify the length upon assembly).